crypto/provider_core.c

```diff
@@ -432,9 +453,45 @@ static int provider_activate(OSSL_PROVIDER *prov)
             prov->query_operation =
                 OSSL_get_provider_query_operation(provider_dispatch);
             break;
+#ifndef OPENSSL_NO_ERR
+# ifndef FIPS_MODE
+        case OSSL_FUNC_PROVIDER_GET_REASON_STRINGS:
+            p_get_reason_strings =
+                OSSL_get_provider_get_reason_strings(provider_dispatch);
+            break;
+# endif
+#endif
         }
     }
 
+#ifndef OPENSSL_NO_ERR
+# ifndef FIPS_MODE
+    if (p_get_reason_strings != NULL) {
+        const ERR_STRING_DATA *reasonstrings =
+            p_get_reason_strings(prov->provctx);
+        size_t cnt;
+
+        /*
+         * Because ERR_load_strings() patches the array's error number
+         * with the error library number, we need to make a copy of that
+         * array.
+         */
+        cnt = 1;                 /* One for the terminating item */
```

Comment:
should be a for loop.

Reply:
That for loop would look like this:

``` C
        for (cnt = 0; reasonstrings[cnt].id != 0; cnt++)
            ;
```

Empty bodies tend to generate a discussion on what they should look like (`;` or `{}`), and I find them generally ugly.  So I made a while loop to avoid both ugliness and that discussion.

Reply:
you either put "continue" before the semi-colon or you move the cnt++ to the semicolon line.
as written, this looks like java.  (ouch, sorry)

Reply:
Dude, this form was prevalent in C before Java came to be, so the inspiration came the other way...

Reply:
(plus, Java has the same for loop construct as well, so that kind of argument was just...)